src/realm/query.cpp

```diff
@@ -1071,15 +1116,23 @@ OldDateTime Query::minimum_olddatetime(size_t column_ndx, size_t* resultcount, s
 
 Timestamp Query::minimum_timestamp(size_t column_ndx, size_t* return_ndx, size_t start, size_t end, size_t limit)
 {
-    ConstTableView tv;
-    tv = find_all(start, end, limit);
+#if REALM_METRICS
+    std::unique_ptr<MetricTimer> metric_timer = QueryInfo::track(this, QueryInfo::type_Minimum);
+#endif
+    TableView tv(*m_table, *this, start, end, limit);
```

Comment:
Yes the change is required. The previous implementation was calling another public function `find_all()` which results in the query being tracked twice, once for `max`/`min` and once for `find_all`. The changes are essentially just the contents of the `find_all` method so there should be no side effect.

The real fix here is to actually do a proper optimisation for max/min in the `TimestampColumn` class instead of just finding all and calling max/min on the column. That would be outside the scope of this PR though.